packages/grafana-data/src/types/explore.ts

```diff
@@ -24,7 +24,7 @@ export interface ExploreTracePanelState {
 
 export interface SplitOpenOptions<T> {
   datasourceUid: string;
-  query: T;
+  query: T | T[];
```

Comment:
Could we change it to?

```suggestion
  queries: T[];
```

Reply:
So how would the hole `SplitOpenOptions` look like. Because this is external API, so `queries?` would have to be optional, as there are places where `queries` wouldn't be implemented. 
 
So would the resulting interface be this?: 

`export interface SplitOpenOptions<T> {
  datasourceUid: string;
  query: T;
  queries?: T[]; 
`
But the user would have to still fill `query`, but then also `queries`? Or then both `query`  and `queries` could be optional? Or we were also thinking about having `query` as required and `additionalQueries` as optional.

But from all of these ` query: T | T[];` seems like cleanest and most understandable in my case. But maybe you have better idea.

Reply:
I don't think it's truly public API 🤔 This function is passed only within Explore to visualizations (which are only internal). So I guess we could just change the type `query -> queries`. We can also mark `query` as deprecated and remove in G10

Reply:
It is also used in `SplitOpen` which is used in `dataLinks` in `packages/grafana-data/src/utils/dataLinks.ts` which is used in `getFieldLinksForExplore` which is used across many panels.

Reply:
It is also used in `SplitOpen` which is used in `dataLinks` in `packages/grafana-data/src/utils/dataLinks.ts` which is used in `getFieldLinksForExplore` which is used across many panels.

Reply:
Which again, we could update, but I am just saying that because it is part of public API, we are not sure if it is not used somewhere else.

Reply:
So I am happy to do whatever Explore team is the best, but personally I thought this is the best and it doesn't require deprecations, changing of optional parameter to required along the way, breaking changes, .... 🙂

Reply:
tbh i feel like @ivanahuckova solution is a bit more robust. i'm also not a big fan of having `query` accept multiple types, but given the way we expose it it feels the safest approach

Reply:
I'd prefer to do more work and have cleaner API in the future. We choose to do it not because it's easy, but because it's hard :D

Reply:
So just to confirm, the preffered solution would be:
1. add `queries?: T[]` type
2. deprecate `queries?: T` and make it optional and let users know they should use queries
3. Internally use `queries` instead of `query` everywhere
4. In Grafana 10, remove `query?` and add breaking change

Does that sound right? @ifrost

Reply:
Sounds good!

I guess you meant:
> deprecate queries?: T
deprecate `query?: T`

Reply:
yes!

Reply:
https://github.com/grafana/grafana/pull/62245/files